### PR TITLE
[SFN] Add usage analytics for Variable Workflow and JSONata features

### PR DESCRIPTION
**Pull request by gregfurman**
## Motivation
This PR adds analytics  the SFN provider to help ascertain usage of the new JSONata and Variable workflows. 

## Changes
- Add a static analyser that will check for:
  -  Any variable sampling or assigning of variables
  - The query language being used by the state machine definition
- Metrics have been added that will record whether a sfn uses JSONata or variable workflows, respectively. 

## Testing
- Added unit tests to ensure the static analysis is correct.

### CODE REVIEW

File: localstack-core/localstack/services/stepfunctions/usage.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jsonpath_create_counter = UsageCounter("stepfunctions:jsonpath:create")
+
+# Count of StepFunctions being created that use Variable Sampling or the Assign block
+variables_create_counter = UsageCounter("stepfunctions:variables:create")
+
+# Successful invocations (also including expected error cases in line with AWS behaviour)
+invocation_counter = UsageCounter("stepfunctions:invocation")
+
+# Unexpected errors that we do not account for
+error_counter = UsageCounter("stepfunctions:error")
```
Comment (gregfurman) [inline localstack-core/localstack/services/stepfunctions/usage.py:20]: Currently unused. Will be implemented in a follow-up

File: localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        return analyser
+
+    def __init__(self):
+        super().__init__()
+        self.has_jsonata: bool = False
+        self.has_variable_sampling = False
+
+    def visitQuery_language_decl(self, ctx: ASLParser.Query_language_declContext):
+        query_language_mode_int = ctx.children[-1].getSymbol().type
```
Comment (MEPalma) [inline localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py:47]: You can probably prune this logic if the flag is already set

**Review by MEPalma — APPROVED** [review]
Looking good from a SFN standpoint!

File: localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+LOG = logging.getLogger(__name__)
+
+
+class UsageMetricsStaticAnalyser(StaticAnalyser):
+    @staticmethod
+    def process(definition: str) -> "UsageMetricsStaticAnalyser":
+        analyser = UsageMetricsStaticAnalyser()
+        analyser.analyse(definition=definition)
+
+        try:
```
Comment (joe4dev) [inline localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py:19]: Do we need the try/catch here? `has_jsonata` is just a getter and we can expect the counters to work (we're not using try/catch anywhere else)
if we want to be as defensive, then we should rather cover the `analyse` step.

**Review by joe4dev — APPROVED** [review]
yeah data 👍 

I would skip the try/catch on the metric counter unless we have a good reason from the SFN analyser side.

File: localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py (hunk 3 above)
Comment (gregfurman) [reply localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py:19]: We didn't want the metrics collection to be the reason why SFN was failing. Will just surround the whole block in a try-catch rather

Comment (gregfurman) [reply localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py:17]: OK wrapped the whole process in a try-except

File: localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py (hunk 2 above)
Comment (gregfurman) [reply localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py:43]: Done!

**Pull request merged**